common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -106,6 +107,46 @@ static uint8_t wpa_bw_to_beerocks_bw(const std::string &chan_width)
     return (chan_width == "80+80") ? 160 : beerocks::string_utils::stoi(chan_width);
 }
 
+/// @brief figures out hostapd config file name by the
+//  interface name and loads its content
+static prplmesh::hostapd::Configuration load_hostapd_config(const std::string &radio_iface_name)
+{
+    std::vector<std::string> hostapd_cfg_names = {
+        "/tmp/run/hostapd-phy0.conf", "/tmp/run/hostapd-phy1.conf", "/var/run/hostapd-phy0.conf",
+        "/var/run/hostapd-phy1.conf", "/var/run/hostapd-phy2.conf", "/var/run/hostapd-phy3.conf"};
+
+    for (const auto &try_fname : hostapd_cfg_names) {
+        LOG(DEBUG) << "Trying to load " << try_fname << "...";
+
+        if (!beerocks::os_utils::file_exists(try_fname)) {
+            continue;
+        }
+
+        prplmesh::hostapd::Configuration hostapd_conf(try_fname);
+
+        // try loading
+        if (!hostapd_conf.load("interface=")) {
+            LOG(ERROR) << "Failed to load hostapd cofig file: " << hostapd_conf;
+            continue;
+        }
+
+        // check if it is the right one:
+        // we are looking for the line in the vap that declares this vap: interface=radio_iface_name
+        // we could equaly ask hostapd_conf if it has this vap, but there is no such interface
+        // to do this. it should be something like: hostapd_conf.is_vap_exists(radio_iface_name);
+        if (hostapd_conf.get_vap_value(radio_iface_name, "interface") != radio_iface_name) {
+            LOG(DEBUG) << radio_iface_name << " does not exists in " << try_fname;
+            continue;
+        }
+
+        // all is good, return the conf
+        return hostapd_conf;
+    }
+
+    // return an empty one since we couldn't find the
+    return prplmesh::hostapd::Configuration("file not found");
```

Comment:
I truly prefer keeping this simple:
* I don't think a pointer is an option, since there is no real need, we can and should use the stack.
* Configuration itself holds its state and it can be queried. This is consistent in all Configuration's functions, including construction.
The only reason I can see to return bool and somehow pass the configuration as an output parameter is to be consistence with other parts of prplMesh.

Reply:
> I truly prefer keeping this simple:

(rhetorical question) In what sense is 
`return prplmesh::hostapd::Configuration("file not found");`
simpler than
`return false;`
?

> The only reason I can see to return bool and somehow pass the configuration as an output parameter is to be consistence with other parts of prplMesh.

That's a very good reason by itself